crates/hc_sandbox/src/cmds.rs

```diff
@@ -117,33 +117,38 @@ pub struct Existing {
 }
 
 impl Existing {
-    pub fn load(mut self) -> anyhow::Result<Vec<PathBuf>> {
-        let sandboxes = crate::save::load(std::env::current_dir()?)?;
+    /// Determine all sandbox paths to use based on .hc and given options.
+    /// Returns at minimum all paths in self.existing_paths.
+    pub fn load(&self) -> anyhow::Result<Vec<PathBuf>> {
+        let saved_sandboxes = crate::save::load(std::env::current_dir()?)?;
+        let mut provided = self.existing_paths.clone();
         if self.all {
-            // Get all the sandboxes
-            self.existing_paths.extend(sandboxes)
-        } else if self.last && sandboxes.last().is_some() {
-            // Get just the last sandbox
-            self.existing_paths
-                .push(sandboxes.last().cloned().expect("Safe due to check above"));
+            // Add all sandboxes in .hc
+            provided.extend(saved_sandboxes);
+        } else if self.last {
+            // Get the last sandbox in .hc
+            if let Some(last) = saved_sandboxes.last() {
+                provided.push(last.clone());
+            }
         } else if !self.indices.is_empty() {
             // Get the indices
-            let e = self
-                .indices
-                .into_iter()
-                .filter_map(|i| sandboxes.get(i).cloned());
-            self.existing_paths.extend(e);
-        } else if !self.existing_paths.is_empty() {
-            // If there is existing paths then use those
-        } else if sandboxes.len() == 1 {
-            // If there is only one sandbox then use that
-            self.existing_paths
-                .push(sandboxes.last().cloned().expect("Safe due to check above"));
-        } else if sandboxes.len() > 1 {
-            // There is multiple sandboxes, the use must disambiguate
+            for i in self.indices.clone() {
```

Comment:
As noted below, this will only catch out of range indices and it won't catch missing paths. So `hc s run 2` in my example below would be caught by this code but it wouldn't tell me that `hc s run 1` isn't pointing to the sandbox that I think it is.

Reply:
Its just below at line 155. It aborts if any requested index points to an invalid path. It will not abort if there is any invalid path at non-requested indices.